ext/search_kit/Civi/Api4/Action/SearchDisplay/ResultDataTrait.php

```diff
@@ -140,10 +162,25 @@ private function outputSpreadsheet(array $rows, array $columns) {
    *
    * @param array $col
    * @param array $value
-   * @return string
+   * @return scalar|null
    */
   protected function formatColumnValue(array $col, array $value) {
-    $val = $value['val'] ?? '';
+    $val = $value['val'];
+
+    if (!in_array($this->format, ['array', 'csv'], TRUE)) {
+      if ($col['rewrite']) {
+        return $val;
+      }
+
+      if ($col['dataType'] === 'Money') {
```

Comment:
This gives php errors when `dataType` doesn't exist.

In general the PR makes excel downloads noticeably slower, but it's still usable.

But the currency does make the field unsortable, and excel doesn't recognize it as a number anymore. Before it included a symbol which it probably shouldn't, but Excel seems ok with that, but after it's now text.

BEFORE:
```
$9.25
$9.25
$44.00
$47.00
$566.00
$12.00
$225.00
$76.00
$133.00
$12.34
```

AFTER:
```
USD, 9.25
USD, 9.25
USD, 44
USD, 47
USD, 566
USD, 12
USD, 225
USD, 76
USD, 133
USD, 12.34
```

Reply:
Thanks for the feedback. Obviously when the download is not triggered from an actual `SearchDisplay` (e.g. from the preview in the "Search For" tab) `dataType` is not set in the call of `formatColumnValue()`. When the download is triggered from an actual `SearchDisplay` it works as expected.

Is it intended that `dataType` is missing when a download is not triggered from a `SearchDisplay`, @colemanw?

Reply:
@dontub good question. The problem is that `$col['dataType']` is deprecated and should not be used. Instead, look up the data type as is done elsewhere. e.g.
https://github.com/civicrm/civicrm-core/blob/662963ad3456c31e92d51b7525a384d3d2f0a8c8/ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php#L218

Reply:
Ok. Are you fine with [this](https://github.com/civicrm/civicrm-core/pull/32766/commits/dabc64005486f6161ed34deaf9e242c2e754affc) solution?

Should `dataType` be removed from `SearchDownloadTest` then?

Reply:
I've removed `dataType` from the xlsx tests, too.